hypothesis/lms
# Store sections on the Grouping table

# Testing notes

- Truncate your courses and grouping tables to test the insertiong of a new course & sections
```truncate grouping, course;```

- Open an assignment with sections https://hypothesis.instructure.com/courses/125/assignments/873

- Check the grouping table, it should contain one row for the course and for each section. Pay attention to the parent_id relationship, sections should point to the course.

```
created                 | 2021-06-01 09:55:54.832308
updated                 | 2021-06-01 09:55:54.832308
id                      | 12
application_instance_id | 8
authority_provided_id   | 653e9620a656a954c684942f6443fa3e3410c03a
parent_id               | 
lms_id                  | f3cd019017839c4630358662a05540f2f6ec5f93
lms_name                | Developer Test Course with Sections Enabled
type                    | course
settings                | {"canvas": {"groups_enabled": true, "sections_enabled": true}, "blackboard": {"files_enabled": true}}
extra                   | {"canvas": {"custom_canvas_course_id": "125"}}
-[ RECORD 2 ]-----------+------------------------------------------------------------------------------------------------------
created                 | 2021-06-01 09:55:55.001108
updated                 | 2021-06-01 09:56:15.790005
id                      | 13
application_instance_id | 8
authority_provided_id   | d2871e5e1c36b58f207d9ef5b22dcb7829f52e61
parent_id               | 12
lms_id                  | 144
lms_name                | Section 1
type                    | canvas_section
settings                | {}
extra                   | null
-[ RECORD 3 ]-----------+------------------------------------------------------------------------------------------------------
created                 | 2021-06-01 09:55:55.001108
updated                 | 2021-06-01 09:56:15.790005
id                      | 14
application_instance_id | 8
authority_provided_id   | f453c6124e4ead4471896f26e896c3e5bbf0bb36
parent_id               | 12
lms_id                  | 145
lms_name                | Section 2
type                    | canvas_section
settings                | {}
extra                   | null
-[ RECORD 4 ]-----------+------------------------------------------------------------------------------------------------------
created                 | 2021-06-01 09:55:55.001108
updated                 | 2021-06-01 09:56:15.790005
id                      | 15
application_instance_id | 8
authority_provided_id   | 83e64baad215f63057387910a1b41debc716a150
parent_id               | 12
lms_id                  | 146
lms_name                | Section 3
type                    | canvas_section
settings                | {}
extra                   | null
```

- Reopen the assignment, not new sections or courses rows added.
- Open a non-sections assignment https://hypothesis.instructure.com/courses/124/assignments/868 there's should be a new course row but not new sections.
